R/preprocessing.R

```diff
@@ -646,7 +646,70 @@ Load10X_Spatial <- function (
   return(object)
 }
 
+#' Load a 10x Genomics Single Cell Experiment into a \code{Seurat} object
+#'
+#' @inheritParams Read10X
+#' @inheritParams SeuratObject::CreateSeuratObject
+#' @param data.dir Directory containing the H5 file specified by \code{filename}
+#' and the image data in a subdirectory called \code{spatial}
+#' @param filename Name of H5 file containing the feature barcode matrix
+#' @param filter.matrix Only keep spots that have been determined to be over
+#' tissue
+#' @param to.upper Converts all feature names to upper case. This can provide an
+#' approximate conversion of mouse to human gene names which can be useful in an
+#' explorative analysis. For cross-species comparisons, orthologous genes should
+#' be identified across species and used instead.
+#' @param ... Arguments passed to \code{\link{Read10X_h5}}
+#'
+#' @return A \code{Seurat} object
+#'
+#'
+#' @export
+#' @concept preprocessing
+#'
+#' @examples
+#' \dontrun{
+#' data_dir <- 'path/to/data/directory'
+#' list.files(data_dir) # Should show filtered_feature_bc_matrix.h5
+#' Load10X(data.dir = data_dir)
+#' }
+#'
+Load10x <- function(data.dir, filename = "filtered_feature_bc_matrix.h5",
+                    assay = "RNA", filter.matrix = TRUE, to.upper = FALSE, ...) {
+
+  if (length(data.dir) > 1) {
+    stop("`data.dir` expects a single directory path but received multiple values.")
+  }
+  if (!file.exists(data.dir)) {
+    stop(paste0("No such file or directory: '", data.dir, "'"))
+  }
+
+
+  filename <- list.files(data.dir, filename, full.names = FALSE, recursive = FALSE)
+  counts.path <- file.path(data.dir, filename)
+  if (!file.exists(counts.path)) {
+    stop(paste0("File not found: '", counts.path, "'"))
+  }
 
+  counts <- Read10X_h5(counts.path, ...)
+  if (is.list(counts)) {
+    object <- Reduce(function(x, y) merge(x, y, add.cell.ids = names(counts)),
```

Comment:
You merge the objects two at a time in the anonymous function here, but the length of `names(counts)` is the length of `counts`, which could be greater than two.

`SeuratObject:::merge.Seurat` does recurse if you give it a single Seurat object in the first argument and a list of Seurat objects in the second. A common construction is to do something like `merge(x[[1]], x[-1], add.cell.ids = names(x)` on a list of Seurat objects `x`.